lib/plug/csrf_protection.ex

```diff
@@ -110,6 +112,21 @@ defmodule Plug.CSRFProtection do
     end
   end
 
+  @doc """
+  Gets the CSRF token with an associated host.
+  """
+  def get_csrf_token_for(host) do
+    token = unmasked_csrf_token()
+    message = generate_token() <> host
+
+    if secret = Process.get(:secret_key_base) do
+      key = KeyGenerator.generate(secret, token)
+      MessageVerifier.sign(message, key)
+    else
+      raise "Process dictionary must contain :secret_key_base."
```

Comment:
We use no trailing period in error messages and we start with a lowercase letter.